src/main/java/net/dries007/tfc/compat/jade/common/BlockEntityTooltips.java

```diff
@@ -0,0 +1,332 @@
+                if (recipe != null)
+                {
+                    tooltip.accept(recipe.getTranslationComponent());
+                    tooltip.accept(Helpers.translatable("tfc.jade.sealed_date").append(ICalendar.getTimeAndDate(Calendars.CLIENT.ticksToCalendarTicks(barrel.getSealedTick()), Calendars.get(level).getCalendarDaysInMonth())));
+                }
+            }
+        }
+    };
+
+    public static final BlockEntityTooltip BELLOWS = (level, state, entity, tooltip) -> {
+        if (entity instanceof BellowsBlockEntity bellows)
+        {
+            int pushTicks = bellows.getTicksSincePushed();
+            if (pushTicks < 20)
+            {
+                pushTicks *= 2;
+                if (pushTicks > 20)
+                {
+                    pushTicks = 40 - pushTicks;
+                }
+
+                MutableComponent component = Helpers.translatable("tfc.jade.bellows_start");
+                String middle = Helpers.translatable("tfc.jade.bellows_middle").getString(); // probably better to translate this once.
+                for (int i = 0; i < pushTicks; i++)
+                {
+                    component.append(Helpers.literal(middle));
+                }
+                component.append(Helpers.translatable("tfc.jade.bellows_end"));
+                tooltip.accept(component);
+            }
+        }
+    };
+
+    public static final BlockEntityTooltip BLAST_FURNACE = (level, state, entity, tooltip) -> {
+        if (entity instanceof BlastFurnaceBlockEntity furnace)
+        {
+            furnace.getCapability(HeatCapability.BLOCK_CAPABILITY).ifPresent(cap -> heat(tooltip, cap.getTemperature()));
+
+            tooltip.accept(Helpers.translatable("tfc.jade.input_stacks", String.valueOf(furnace.getInputCount())));
+            tooltip.accept(Helpers.translatable("tfc.jade.catalyst_stacks", String.valueOf(furnace.getCatalystCount())));
+            tooltip.accept(Helpers.translatable("tfc.jade.fuel_stacks", String.valueOf(furnace.getFuelCount())));
+        }
+    };
+
+    public static final BlockEntityTooltip BLOOMERY = (level, state, entity, tooltip) -> {
+        if (entity instanceof BloomeryBlockEntity bloomery && state.getBlock() instanceof BloomeryBlock)
+        {
+            tooltip.accept(Helpers.translatable("tfc.jade.input_stacks", bloomery.getInputStacks().size()));
+            tooltip.accept(Helpers.translatable("tfc.jade.catalyst_stacks", bloomery.getCatalystStacks().size()));
+            if (state.getValue(BloomeryBlock.LIT))
+            {
+                final long ticksLeft = bloomery.getRemainingTicks();
+                if (ticksLeft > 0)
+                {
+                    final BloomeryRecipe recipe = bloomery.getCachedRecipe();
+                    if (recipe != null)
+                    {
+                        tooltip.accept(Helpers.translatable("tfc.jade.hours_remaining", String.valueOf(Mth.ceil((float) ticksLeft / ICalendar.TICKS_IN_HOUR))));
```

Comment:
This tooltip should use `Calendars.get(level).getTimeDelta(ticksLeft)`. Standardization!